docs/dev/table/sourceSinks.zh.md

```diff
@@ -193,6 +192,149 @@ for more information.
 The runtime implementation of a `LookupTableSource` is a `TableFunction` or `AsyncTableFunction`. The function
 will be called with values for the given lookup keys during runtime.
 
+#### Defining a Dynamic Table Source with Projection Push-Down
```

Comment:
How about we just describe the interfaces in 1 or 2 sentences. Maybe in a table. And link to the corresponding class on Github. It is very difficult to keep docs and interfaces in sync. We should avoid code duplication.